### PR TITLE
fix 1d/2d array method error message with unsupported keyword arg

### PR DESCRIPTION
**Pull request by Moelf**
based on #36321

### CODE REVIEW

File: base/errorshow.jl [hunk 1]
```diff
@@ -315,7 +315,7 @@ function showerror(io::IO, ex::MethodError)
             hasrows |= isrow
             push!(vec_args, isrow ? vec(arg) : arg)
         end
-        if hasrows && applicable(f, vec_args...)
+        if hasrows && applicable(f, vec_args...) && isempty(kwargs)
```
Comment (mbauman) [inline base/errorshow.jl:318]: It'd be awesome if we could do:

```julia
        if hasrows && applicable(f, vec_args...; kwargs...)
```

but looks like applicable doesn't understand kwargs.  This seems like a good enough stop-gap.

**Comment by Moelf** [comment]
bump

**Comment by StefanKarpinski** [comment]
Who would be good to review this?

**Comment by mbauman** [comment]
This is a decent band-aid, IMO.  It'd be easier to do something better with the just-filed #36611 or by transforming the usage here to call `hasmethod` instead, but I think the current workaround is good enough (even if not ideal)

It'd be nice to add a few simple tests, though.

**Comment by Moelf** [comment]
tests added ;)

**Comment by Moelf** [comment]
not sure why doctest failed; anything else needed?

**Comment by Moelf** [comment]
bump for 1.6 train

**Review by mbauman — APPROVED** [review]

File: test/errorshow.jl [hunk 2]
```diff
@@ -750,3 +750,12 @@ end
     @test contains(bt_str, "@ $m.B.C")
     @test contains(bt_str, "@ $m.B.D")
 end
+# 1d/2d error shouldn't appear in unsupported keywords arg #36325
+let err = nothing
+    try
+        identity([1,1]; bad_kwards = :julia)
```
Comment (JeffBezanson) [inline test/errorshow.jl:756]:
```suggestion
        identity([1 1]; bad_kwards = :julia)
```

**Pull request merged**